### DISCUSSION TITLE
Adding feeds which require authentication

### DISCUSSION THREAD

**Discussion by brockhaywood** [transitland/transitland#354] [Q&A] [answered]
Hi there,

I'm looking to add a new feed to transitland atlas but I'm unclear on how to add auth params. In particular, an agency which I want to add has a require to fill out a form to get the URL and api key. I've read the following:

https://github.com/transitland/transitland-atlas#how-to-add-a-new-feed

The part that is unclear or I missed is where does the secret get persisted and where an agreement with the agency needs to be signed to get the API key, who can perform that?

**Comment by drewda** [comment, marked as answer]
Hi @brockhaywood,

There are some examples of Transitland Atlas records with auth information in https://github.com/transitland/transitland-atlas/blob/master/feeds/api.goswift.ly.dmfr.json

Full info about the auth options are in the DMFR specification at https://github.com/transitland/distributed-mobility-feed-registry/blob/master/json-schema/dmfr.schema-v0.4.0.json#L179-L197

For the deployment of Transitland running at transit.land, Interline agrees to terms and maintains a private list of secrets. If you are representing an agency, you are welcome to contact us at info@interline.io about any next steps for signing up for access keys.

**Reply by brockhaywood** [reply]
Awesome. This is helpful, thanks!
